apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.Options.cs

```diff
@@ -94,23 +108,40 @@ private Options(TimeSpan? duration, DateTimeOffset? expiration, SigningVersion v
             /// <param name="duration">The new duration.</param>
             /// <returns>A new set of options with the given duration.</returns>
             public Options WithDuration(TimeSpan duration) =>
-                new Options(duration, null, SigningVersion);
+                new Options(duration, null, SigningVersion, UrlStyle, Scheme);
 
             /// <summary>
             /// Returns a new set of options with the same values as this one but expiration based.
             /// </summary>
             /// <param name="expiration">The new expiration.</param>
             /// <returns>A new set of options with the given expiration.</returns>
             public Options WithExpiration(DateTimeOffset expiration) =>
-                new Options(null, expiration, SigningVersion);
+                new Options(null, expiration, SigningVersion, UrlStyle, Scheme);
 
             /// <summary>
             /// Returns a new set of options with the same values as this one except for the signing version.
             /// </summary>
             /// <param name="version">The new signing version.</param>
             /// <returns>A set of options with the given signing version.</returns>
             public Options WithSigningVersion(SigningVersion version) =>
-                new Options(Duration, Expiration, GaxPreconditions.CheckEnumValue(version, nameof(version)));
+                new Options(Duration, Expiration, GaxPreconditions.CheckEnumValue(version, nameof(version)), UrlStyle, Scheme);
+
+            /// <summary>
+            /// Returns a new set of options with the same values as this one except for the
+            /// <see cref="UrlStyle"/> value.
+            /// </summary>
+            /// <param name="urlStyle">The new url style.</param>
+            /// <returns>A new set ofoptions with the given url style.</returns>
+            public Options WithUrlStyle(UrlStyle urlStyle) =>
+                new Options(Duration, Expiration, SigningVersion, urlStyle, Scheme);
+
+            /// <summary>
+            /// Returns a new set of options with the same values as this one except for the scheme.
+            /// </summary>
+            /// <param name="scheme">The new scheme.</param>
+            /// <returns>A new set of options with the given scheme.</returns>
+            public Options WithScheme(string scheme) =>
```

Comment:
Is there anything other than http or https that would work? If not, should we validate that? Also document the null behavior.

Reply:
Pending discussion from @houglum, but I'm fairly certain these are the only two values for scheme.